modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogManagerImpl.java

```diff
@@ -418,6 +440,15 @@ public List<SystemView<?>> systemViews() {
     class OnUpdateHandlerImpl implements OnUpdateHandler {
         @Override
         public CompletableFuture<Void> handle(VersionedUpdate update, HybridTimestamp metaStorageUpdateTimestamp, long causalityToken) {
+            if (update instanceof SnapshotUpdate) {
+                Catalog catalog = ((SnapshotUpdate) update).snapshotEntry().applyUpdate(catalogByVer.get(0), causalityToken);
```

Comment:
after the first compaction the catalog of version 0 will be truncated as well. This is kinda confusing (I mean, we definitely will get null here, why do we even do this in the first place?)